tap_core/src/tap_manager/manager.rs

```diff
@@ -135,6 +135,27 @@ impl<
         Ok(())
     }
 
+    /// Removes all receipts that have a timestamp less or equal than the last RAV timestamp_max.
+    /// No-op if there is no last RAV.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`Error::AdapterError`] if there are any errors while retrieving last RAV or removing receipts
+    ///
+    pub fn remove_receipts_older_than_last_rav(&mut self) -> Result<(), Error> {
```

Comment:
also name can be confusing since it says older than but it does less or equal

Reply:
> edit: Might even return the deleted receipts in case user wants to log them or something. But I guess this might require changes in existing functions so maybe we can add it later

I am refraining from returning too many things there, because that will constrain the library user's design space. Imagine if the user want to implement the storage in a DB. Then returning receipts might incur an undesired performance cost. Ideally, the user could choose what's returned (instead of being constrained by the trait), but I don't know how to do that elegantly here. Perhaps there is a roundabout way possible today, in the implementation of the ReceiptStorageAdapter trait.

> also name can be confusing since it says older than but it does less or equal

Actually, I was wondering if there was a more concise way to express that it deletes the receipts that are too old to be aggregated with the last RAV.

Reply:
Makes sense about returning stuff.

I can't come up with better naming either so I think this is just nitpicking, no need to waste time on it I guess since we can change it later as well.

thanks!

Reply:
For the function name I think it should just point out that the receipts are already aggregated into the RAV. Something like `remove_aggregated_receipts` or `remove_receipts_aggregated_in_rav` or something to that affect. The main thing we want the user calling the function to know is that they are deleting obsolete receipts.

I agree that it would be nice to allow the receipts to be archived, but I think that feature already exist. Keep in mind the user is implementing the `remove_receipts_in_timestamp_range` function. They can easily have that function just delete from one key in the DB and move it to another or some other approach to archiving them. So we kind of get the flexibility we are hoping for (users can either delete or archive receipts) for free.

Reply:
The thing though is that the function does not check that the receipts have really been aggregated. Not sure if that could happen (perhaps with "failed receipts"), but this function will happily remove non-aggregated receipts if they are older than the latest RAV, if that ever happens.